src/client/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -96,6 +97,13 @@ private void AddIosSpecificParametersToPayload(Dictionary<string, string> broker
             {
                 _brokerRequestNonce = Guid.NewGuid().ToString();
                 brokerPayload[iOSBrokerConstants.BrokerNonce] = _brokerRequestNonce;
+
+                string applicationToken = ReadBrokerApplicationTokenFromKeychain(brokerPayload);
+
+                if (!string.IsNullOrEmpty(applicationToken))
+                {
+                    brokerPayload.Add(iOSBrokerConstants.ApplicationToken, applicationToken);
```

Comment:
>brokerPayload.Add [](start = 20, length = 17)

Can there be more than one application token in the payload? If not any reason to not assign the value directly in the dictionary?